src/ImageGallery.jsx

```diff
@@ -783,17 +867,24 @@ export default class ImageGallery extends React.Component {
 
       const showItem = !this.props.lazyLoad || alignment || this._lazyLoaded[index];
       if (showItem && this.props.lazyLoad) {
-          this._lazyLoaded[index] = true;
+        this._lazyLoaded[index] = true;
       }
 
+      const imageSize = this.props.lazyLoad && this._isItemLazyLoadedSuccessfully[index] !== true
+        ? {
+            width: this.state.gallerySlideWrapperWidth,
+            height: this.state.gallerySlideWrapperHeight
+          }
+        : {};
+
       const slide = (
         <div
           key={index}
           className={'image-gallery-slide' + alignment + originalClass}
-          style={Object.assign(this._getSlideStyle(index), this.state.style)}
+          style={Object.assign(this._getSlideStyle(index), this.state.style, imageSize)}
           onClick={this.props.onClick}
         >
-          {showItem ? renderItem(item) : <div style={{ height: '100%' }}></div>}
+          {showItem ? renderItem(item, this._onImageLoad.bind(this, index)) : <div style={{ height: '100%' }}></div>}
```

Comment:
Now we need to know which items loaded successfully. (If an item was not loaded we should set a size of the latest active element for the container of this item.)
So all custom renderers should call the onImageLoad function or the image will be limited in height and width. (Only in lazyload mode)